tsconfig.json

```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
+    "allowJs": false,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "incremental": true,
+    "isolatedModules": true,
+    "jsx": "react-jsx",
+    "module": "ES2020",
+    "moduleResolution": "node",
+    "noEmit": false,
+    "noImplicitReturns": true,
+    "resolveJsonModule": false,
+    "rootDir": "./src",
+    "skipLibCheck": true,
+    "strict": true,
+    "target": "ES2022",
```

Comment:
What versions of Next, React, browser are we targeting? Is that resolved by the user of this package? In that case, are we limiting their options?

Reply:
This option is related to how "old" is the javascript output generated. It is not related to the Next or React versions.

From [this compatibility table](https://compat-table.github.io/compat-table/es2016plus/) we can check the browsers, but it seems to me all relevant features from this package are already supported in ES2022 by all browsers (They are even for ES2023, but I guess we could be a bit _conservative_).

Lowering the target may end up in larger bundles in javascript, as more functions/APIs/syntax needs to be traspiled by Typescript to Javascript to enable support for those browsers, if needed.

We can talk about it on monday